### PR TITLE
Fix passphrase missmatch refactoring2

### PR DESCRIPTION
**Pull request by peter-sanderson**
## Related Issue

Resolve

### CODE REVIEW

File: suite-common/wallet-core/src/device/deviceThunks.ts [hunk 1]
```diff
@@ -412,17 +427,25 @@ export const authConfirm = createThunk(
...
 
-                if (settings.isViewOnlyModeVisible) {
+                if (
+                    response.payload.error === 'auth-confirm-retry' &&
+                    settings.isViewOnlyModeVisible
+                ) {
                     const newDevice = selectDeviceSelector(getState());
                     dispatch(deviceActions.selectDevice(newDevice));
+                    extra.thunks.onPassphraseConfirmRetry();
+                    // dispatch(addWalletThunk({ walletType }));
```
Comment (peter-sanderson) [inline suite-common/wallet-core/src/device/deviceThunks.ts]: :red_circle:

File: suite-common/wallet-core/src/device/deviceThunks.ts [hunk 2]
```diff
@@ -378,6 +378,27 @@ export const authorizeDeviceThunk = createThunk<
...
+            response.payload.error === 'enter-passphrase-back'
+        ) {
+            const settings = extra.selectors.selectSuiteSettings(getState());
+            dispatch(deviceActions.forgetDevice({ device, settings }));
+
+            if (settings.isViewOnlyModeVisible) {
+                const newDevice = selectDeviceSelector(getState());
+                dispatch(deviceActions.selectDevice(newDevice));
+                if (response.payload.error === 'enter-passphrase-back') {
+                    dispatch(extra.thunks.onPassphraseConfirmRetry());
```
Comment (peter-sanderson) [inline suite-common/wallet-core/src/device/deviceThunks.ts]: retry in back condition is weird

File: packages/suite/src/components/wallet/DiscoveryProgress.tsx [hunk 3]
```diff
@@ -6,7 +6,7 @@ import { zIndices } from '@trezor/theme';
 
 const StyledProgressBar = styled(ProgressBar)`
     height: 0;
-    z-index: ${zIndices.discoveryProgress};
```
Comment (peter-sanderson) [inline packages/suite/src/components/wallet/DiscoveryProgress.tsx]: is this ever used after we remove it from here?

File: packages/suite/src/components/wallet/DiscoveryProgress.tsx [hunk 4]
```diff
@@ -6,7 +6,7 @@ import { zIndices } from '@trezor/theme';
...
+    z-index: ${zIndices.modal};
```
Comment (peter-sanderson) [inline packages/suite/src/components/wallet/DiscoveryProgress.tsx]: Not sure about this

image

**Review by jvaclavik — APPROVED** [review]

**Pull request merged**